doc/comptime.org

```diff
@@ -0,0 +1,495 @@
+#+bibliography: bibliography.bib
+
+* Comptime
+Zig has a concept called ~comptime~, it's stands for "compile-time".
+Comptime is used to evaluate an expression at compile time and not at runtime.
+In comparison with C, Zig comptime has the purpose of replacing marco with a more explicit syntax.
+In fact, C's macro tends to be error-prone when using it.
+The advantage of using comptime over macro is the type safety of Zig when writing comptime.
+
+** Compile-time evaluation
+*** Compile-time variable
+
+In Zig, there are variables that can be evaluated at compile-time, in fact, Zig allows computing mutating variables at compile-time but all the inputs need to be known also at compile-time.
+If not, the compiler will throw a compile error. [cite:@DocumentationZigProgramming]
+
+For a mutating variable at compile-time, Zig requires naming the variables with a ~comptime var~.
+But if the variable is a constant, the compiler requires to use a ~const~. 
+
+Like in the example below, the variable named ~variableAtCompileTime~ is evaluated at compile-time because all the inputs are known.
+On the other hand, the variable named ~constantAtRuntime~ cannot be a comptime variable because its dependency is based on unknown before runtime.
+
+Moreover, in the example, the ~inline for~ is used to unroll the for loop.
+This allows to use for loops in comptime indeed, if a standard for loop is used, it will cause an error because the capture value will be evaluated at runtime.[cite:@DocumentationZigProgramming] 
+#+begin_src zig :imports '(std) :exports both
+  const stdout = std.io.getStdOut().writer();
+
+  const randVariable = std.crypto.random.float(f32);
+  const selectedConstant = 6;
+
+  const constantAtRuntime = randVariable * selectedConstant;
+
+  comptime var variableAtCompileTime = selectedConstant * selectedConstant;
+  const array = [_]comptime_int { 3, 2, 1};
+
+  inline for (array) |item| {
+      variableAtCompileTime += item;
+  }
+
+  try stdout.print("constant-at-runtime {d:.2}\n", .{constantAtRuntime});
+  try stdout.print("variable-at-compile-time  {d}", .{variableAtCompileTime});
+
+#+end_src
+
+#+RESULTS:
+| constant-at-runtime      | 0.0 |
+| variable-at-compile-time |  42 |
+
+*** Compile-time expression
+In Zig, an expression can have an ~comptime~ to tell the compiler to evaluate the expression at compile-time.
+Like a compile-time variable, if an expression cannot be evaluated at compile-time, a compile-time error will be thrown.
+
+With a prefixed ~comptime~ keyword Zig can interpret a function at compile-time instead of runtime. [cite:@DocumentationZigProgramming]
+
+A good example of demonstrating comptime expression is in the standard documentation [cite:@DocumentationZigProgramming].
+The results show that the comptime expression is faster than the runtime one.
```

Comment:
I dont think that comptime expression are necessercly "faster" than runtime ones, its just that they are done at compilation time, but in my opinion the time taken is not relly comparable.
However I think it is more relvant to say that our "real running application" is going to be faster since work has already be done at compile time

Reply:
I think that in your exemple the timer records "nothing" except doing a *store* to *result*